FeaturedArticleWidget/FeaturedArticleWidget.swift

```diff
@@ -94,35 +81,65 @@ class FeaturedArticleWidget: ExtensionViewController, NCWidgetProviding {
         expandedArticleView.tintColor = theme.colors.link
     }
     
+    var articleURL: URL?
+    
     func widgetPerformUpdate(completionHandler: @escaping (NCUpdateResult) -> Void) {
-        defer {
-            updateView()
+        WidgetController.shared.startWidgetUpdateTask(completionHandler) { (dataStore, completion) in
+            let moc = dataStore.viewContext
+            let siteURL = dataStore.languageLinkController.appLanguage?.siteURL()
+            moc.perform {
+                guard let featuredContentGroup = moc.newestGroup(of: .featuredArticle, forSiteURL: siteURL),
+                    let articleURL = featuredContentGroup.contentPreview as? URL else {
+                    completionHandler(.noData)
+                    return
+                }
+                let article = moc.fetchArticle(with: articleURL)
+                self.update(with: article, dataStore: dataStore) { result in
+                    completionHandler(result ? .newData : .noData)
```

Comment:
I think you need to run `completion` here, not `completionHandler`? I get a bit confused on when each needs to be called.

Reply:
That's right, good catch!